encoder.go

```diff
@@ -170,9 +179,9 @@ func (e *GenericTemplateBasedEncoder) Files() []*plugin_go.CodeGeneratorResponse
 	resultChan := make(chan *plugin_go.CodeGeneratorResponse_File, length)
 	for _, templateFilename := range templates {
 		go func(tmpl string) {
-			content, translatedFilename, err := e.buildContent(tmpl)
-			if err != nil {
-				errChan <- err
+			content, translatedFilename, err2 := e.buildContent(tmpl)
```

Comment:
Not sure about how to do it properly, should I reuse the first `err` and create empty `var` variables for `content` and `translatedFilename` to switch the `:=` with `=` or is it better to keep `err2` in this case?

cc @abronan

Reply:
Simply doing `content, translatedFilename, err := ... ` should just work, but maybe creating a new variable (like you did) that doesn't escape the closure is cleaner ?

Reply:
using `content, translatedFilename, err := ...` triggers a variable shadowing lint error, I'm declaring variable and reusing previous `err`, it fixes the lint warns, keeps the code readable, and works